crates/deltalake-core/tests/integration.rs

```diff
@@ -61,23 +61,34 @@ async fn test_action_reconciliation() {
     };
 
     assert_eq!(2, fs_common::commit_removes(&mut table, vec![&r]).await);
-    assert_eq!(table.get_files_iter().count(), 0);
+    assert_eq!(table.get_files_iter().unwrap().count(), 0);
     assert_eq!(
         table
-            .get_state()
-            .all_tombstones()
-            .iter()
-            .map(|r| r.path.as_str())
+            .snapshot()
+            .unwrap()
+            .all_tombstones(table.object_store().clone())
+            .await
+            .unwrap()
+            .map(|r| r.path.clone())
             .collect::<Vec<_>>(),
-        vec![a.path.as_str()]
+        vec![a.path.clone()]
     );
```

Comment:
While this part of the test does not pass anymore, i feel this is the desired behavior.

What this tests, if we can add an already deleted file back into the table, using the same path etc. i.e. un-doing a remove. The thing to do would be to time travel, or add the same data under a new file name...